src/com/cdd/bao/Main.java

```diff
@@ -235,7 +236,23 @@ private static void compileSchema(String[] options) throws Exception
 		Util.writeln("Loaded: " + schvoc.numTerms() + " terms.");
 		try (OutputStream ostr = new FileOutputStream(outputFile))
 		{
-			schvoc.serialise(ostr);
+			if (outputFile.endsWith(".json"))
+			{
+				JSONArray templates = new JSONArray();
+				for (Schema schema : schemata)
+				{
+					SchemaVocab subset = schvoc.singleTemplate(schema.getSchemaPrefix());
+				
+					JSONObject json = new JSONObject();
+					json.put("schemaPrefix", schema.getSchemaPrefix());
+					json.put("root", ClipboardSchema.composeGroup(schema.getRoot(), subset));
+					templates.put(json);
+				}
+				Writer wtr = new BufferedWriter(new OutputStreamWriter(ostr));
+				templates.write(wtr);
+				wtr.flush();
+			}
```

Comment:
Why have this here and not where the normal serialisation code is. That would be the more logical place.

Reply:
It's not a mainstream feature, so I don't want to make it too readily accessible.